juriscraper/pacer/utils.py

```diff
@@ -296,3 +299,32 @@ def get_pdf_url(court, pacer_case_id, document_number, attachment_number):
             court, pacer_case_id, document_number, attachment_number
         ),
     )
+
+
+def parse_datetime_for_us_timezone(datetime_str: str) -> datetime:
+    """Parse a datetime from a string that contains a US timezone.
+
+    :param datetime_str: The str datetime to parse.
+    :return: A datetime object with UTC timezone offset.
+    """
+
+    # Supported US timezones.
+    tzinfos = {
```

Comment:
I worry that this isn't enough if this is used in district/bankruptcy courts. For example, the Northern Mariana Islands are in Chamorro Standard Time...

Reply:
I'm surprised there's not a way to just do *all* of the timezones via something simple?

Reply:
Yeah, I have done some research on this topic, and seems there's not a simpler way to handle timezones properly.
There is a library `pytz` that has some useful methods to retrieve timezones by country.

For US the list is the following:
```
America/New_York
America/Detroit
America/Kentucky/Louisville
America/Kentucky/Monticello
America/Indiana/Indianapolis
America/Indiana/Vincennes
America/Indiana/Winamac
America/Indiana/Marengo
America/Indiana/Petersburg
America/Indiana/Vevay
America/Chicago
America/Indiana/Tell_City
America/Indiana/Knox
America/Menominee
America/North_Dakota/Center
America/North_Dakota/New_Salem
America/North_Dakota/Beulah
America/Denver
America/Boise
America/Phoenix
America/Los_Angeles
America/Anchorage
America/Juneau
America/Sitka
America/Metlakatla
America/Yakutat
America/Nome
America/Adak
Pacific/Honolulu
```
However, some timezones are not included like:
Pacific/Guam
Pacific/Samoa
America/Puerto_Rico

In addition, a mapping between these timezones and their code is needed like `EST -> America/New_York`

So that since we are already using `dateutil` I think this is the simpler way to handle timezones.

I've added the timezones above to the list:

```
"CHST":"Pacific/Guam"
"SST": "Pacific/Samoa"
"AST": "America/Puerto_Rico"
```
So according to [Wikipedia](https://en.wikipedia.org/wiki/Time_in_the_United_States) now we have all the US timezones.

But I'm afraid that we might need an additional mapping between time zone codes and courts. I was checking court timezones in CL and I'm surprised that some courts use a local timezone e.g: `"arb": "US/Arizona"`, while Arizona uses MST time, which I mapped as `America/Denver`, so not sure if some of these courts use a different timezone according if some regions of the state use DST or not, do you know if the reason is that?

So far I've found that only appellate courts use local timezones and district courts use GMT, so maybe we don't need to map all the timezone names, I'll confirm this by checking RSS for all district/bankruptcy courts.

Reply:
Arizona is special and if we want to do this function right, MST is not good enough because sometimes Arizona uses MST, but sometimes they use PDT.